com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -432,163 +494,141 @@ void Send(NetworkTransformState stateToSend)
             // making it immobile.
             if (isDirty)
             {
-                Send(m_LocalAuthoritativeNetworkState);
+                // Commit the state
+                ReplicatedNetworkState.Value = m_LocalAuthoritativeNetworkState;
                 m_HasSentLastValue = false;
                 m_LastSentTick = m_CachedNetworkManager.LocalTime.Tick;
                 m_LastSentState = m_LocalAuthoritativeNetworkState;
             }
             else if (!m_HasSentLastValue && m_CachedNetworkManager.LocalTime.Tick >= m_LastSentTick + 1) // check for state.IsDirty since update can happen more than once per tick. No need for client, RPCs will just queue up
             {
+                m_LastSentState.IsTeleportingNextFrame = false; // This is required here
                 m_LastSentState.SentTime = m_CachedNetworkManager.LocalTime.Time; // time 1+ tick later
-                Send(m_LastSentState);
+                // Commit the state
+                ReplicatedNetworkState.Value = m_LastSentState;
                 m_HasSentLastValue = true;
             }
         }
 
-        [ServerRpc(RequireOwnership = false)]
-        private void CommitTransformServerRpc(NetworkTransformState networkState, ServerRpcParams serverParams = default)
-        {
-            if (serverParams.Receive.SenderClientId == OwnerClientId) // RPC call when not authorized to write could happen during the RTT interval during which a server's ownership change hasn't reached the client yet
-            {
-                CommitLocallyAndReplicate(networkState);
-            }
-        }
-
-        private void CommitLocallyAndReplicate(NetworkTransformState networkState)
-        {
-            m_ReplicatedNetworkState.Value = networkState;
-
-            if (Interpolate)
-            {
-                AddInterpolatedState(networkState);
-            }
-        }
-
         private void ResetInterpolatedStateToCurrentAuthoritativeState()
         {
             var serverTime = NetworkManager.ServerTime.Time;
             m_PositionXInterpolator.ResetTo(m_LocalAuthoritativeNetworkState.PositionX, serverTime);
             m_PositionYInterpolator.ResetTo(m_LocalAuthoritativeNetworkState.PositionY, serverTime);
             m_PositionZInterpolator.ResetTo(m_LocalAuthoritativeNetworkState.PositionZ, serverTime);
 
-            m_RotationInterpolator.ResetTo(Quaternion.Euler(m_LocalAuthoritativeNetworkState.Rotation), serverTime);
+            m_RotationInterpolator.ResetTo(Quaternion.Euler(m_LocalAuthoritativeNetworkState.RotAngleX, m_LocalAuthoritativeNetworkState.RotAngleY, m_LocalAuthoritativeNetworkState.RotAngleZ), serverTime);
```

Comment:
I don't know if this really matters that much, but when I was doing FastBufferReader/Writer I had trouble writing tests when I was doing quaternions this way, because quaternion -> euler -> quaternion is not a lossless process. The quaternion would always end up different so I changed it to just send the quaternion directly instead of converting it to euler angles because that was the only way I could get it deterministic enough to test.

This isn't really related to your changes but felt worth mentioning. I don't know enough about quaternions to know whether that loss in the conversion might cause any issues.

Reply:
You do lose resolution but it won't be more than the minimum threshold value to detect change.  
So, we are not 100% lossless but relative to the resolution being sent it works out.